src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -420,14 +437,36 @@ private void WatchForFileSystemEventsThreadStart(object arg)
                             }
                         }
                     }
+
+                    ArraySegment<char> underlyingArray;
+                    if (MemoryMarshal.TryGetArray(events[i], out underlyingArray))
+                        ArrayPool<char>.Shared.Return(underlyingArray.Array);
+                }
+
+                unsafe void ProcessEvents()
+                {
+                    for (int i = 0; i < events.Length; i++)
+                    {
+                        int endPoint = GetIndex(eventPaths[i]);
+                        Debug.Assert(endPoint > 0, "Empty events are not supported");
+                        events[i] = new Memory<char>(ArrayPool<char>.Shared.Rent(endPoint));
```

Comment:
Technically you should use `Encoding.UTF8.GetMaxCharCount(endPoint)`. Also, `byteCount` is a clearer name than `endPoint`.